From: Jason Rhinelander <jason@imaginary.ca>
Date: Mon, 5 Feb 2024 17:11:22 -0400
Subject: Fix libevent linker issue

With this inlined in the header, calling code itself would have to link
against libevent (rather than just transitively via libquic).
---
 include/oxen/quic/utils.hpp | 6 +-----
 src/utils.cpp               | 7 +++++++
 2 files changed, 8 insertions(+), 5 deletions(-)

diff --git a/include/oxen/quic/utils.hpp b/include/oxen/quic/utils.hpp
index 73a33de..b068e12 100644
--- a/include/oxen/quic/utils.hpp
+++ b/include/oxen/quic/utils.hpp
@@ -200,11 +200,7 @@ namespace oxen::quic
 
     struct event_deleter final
     {
-        void operator()(::event* e) const
-        {
-            if (e)
-                ::event_free(e);
-        }
+        void operator()(::event* e) const;
     };
     using event_ptr = std::unique_ptr<::event, event_deleter>;
 
diff --git a/src/utils.cpp b/src/utils.cpp
index c9f6ba0..d130a2d 100644
--- a/src/utils.cpp
+++ b/src/utils.cpp
@@ -37,4 +37,11 @@ namespace oxen::quic
         std::transform(s.begin(), s.end(), s.begin(), [](unsigned char c) { return std::tolower(c); });
         return s;
     }
+
+    void event_deleter::operator()(::event* e) const
+    {
+        if (e)
+            ::event_free(e);
+    }
+
 }  // namespace oxen::quic
